common/txmgr/types/tx.go

```diff
@@ -26,3 +26,8 @@ const (
 	TxAttemptInsufficientEth = TxAttemptState("insufficient_eth")
 	TxAttemptBroadcast       = TxAttemptState("broadcast")
 )
+
+// Transaction is the type that callers get back, when they create a Transaction using the Txm.
+type Transaction interface {
+	GetID() string
```

Comment:
Drop the `Get` prefix?
```suggestion
	ID() string
```

Reply:
Can't do that, since EthTx has a field named ID. Also, this is throwaway function, till I actually extract EthTx out to this namespace with a generic Tx name, and replace this interface. I couldn't do that earlier, due to circular deps.